stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/message/attachments/internal/AttachmentUploader.kt

```diff
@@ -36,34 +47,172 @@ internal class AttachmentUploader(
-            val call = client.sendImage(channelType, channelId, file, progressCallback)
-            call.await()
+        return if (attachmentType == AttachmentType.IMAGE) {
+            uploadImage(
+                channelType = channelType,
+                channelId = channelId,
+                file = file,
+                progressCallback = progressCallback,
+                attachment = attachment,
+                mimeType = mimeType,
+                attachmentType = attachmentType
+            )
         } else {
-            val call = client.sendFile(channelType, channelId, file, progressCallback)
-            call.await()
+            uploadFile(
+                channelType = channelType,
+                channelId = channelId,
+                file = file,
+                progressCallback = progressCallback,
+                attachment = attachment,
+                mimeType = mimeType,
+                attachmentType = attachmentType
+            )
         }
+    }
+
+    /**
+     * Uploads an image attachment.
+     *
+     * @param channelType The type of the channel.
+     * @param channelId The ID of the channel.
+     * @param file The file that will be uploaded.
+     * @param attachment The attachment to be uploaded.
+     * @param progressCallback Used to listen to file upload
+     * progress, success, and failure.
+     * @param mimeType The mime type of the attachment that will be uploaded,
+     * e.g. image/jpeg.
+     * @param attachmentType The type of the attachment, e.g. "video", "audio", etc.
+     *
+     * @return The resulting uploaded attachment.
+     */
+    private suspend fun uploadImage(
+        channelType: String,
+        channelId: String,
+        file: File,
+        progressCallback: ProgressCallback?,
+        attachment: Attachment,
+        mimeType: String,
+        attachmentType: AttachmentType,
+    ): Result<Attachment> {
+        val result = client.sendImage(channelType, channelId, file, progressCallback)
+            .await()
+
         return if (result.isSuccess) {
             val augmentedAttachment = attachment.augmentAttachmentOnSuccess(
                 file = file,
-                mimeType = mimeType ?: "",
+                mimeType = mimeType,
                 attachmentType = attachmentType,
                 url = result.data()
             )
-            augmentedAttachment.uploadState = Attachment.UploadState.Success
-            progressCallback?.onSuccess(augmentedAttachment.url)
-            Result(augmentedAttachment)
+
+            onSuccessfulUpload(
+                augmentedAttachment = augmentedAttachment,
+                progressCallback = progressCallback
+            )
+        } else {
+            onFailedUpload(
+                attachment = attachment,
+                result = result,
+                progressCallback = progressCallback
+            )
+        }
+    }
+
+    /**
+     * Uploads an image attachment.
```

Comment:
Good point. Since it's mostly the same params it's been copy pasted from the `uploadImage`  function, but it looks like I forgot to edit the description.